simtools/production_configuration/calculate_statistical_errors_grid_point.py

```diff
@@ -0,0 +1,460 @@
+                    "bin_edges_high": bin_edges_high,
+                    "simulated_event_histogram": simulated_event_histogram,
+                    "viewcone": viewcone,
+                    "core_range": core_range,
+                }
+                unique_azimuths = np.unique(events_data["PNT_AZ"]) * u.deg
+                unique_zeniths = np.unique(events_data["PNT_ALT"]) * u.deg
+                if self.grid_point is None:
+                    _logger.info(f"Unique azimuths: {unique_azimuths}")
+                    _logger.info(f"Unique zeniths: {unique_zeniths}")
+
+                    if len(unique_azimuths) == 1 and len(unique_zeniths) == 1:
+                        _logger.info(
+                            f"Setting initial grid point with azimuth: {unique_azimuths[0]}"
+                            f" zenith: {unique_zeniths[0]}"
+                        )
+                        self.grid_point = (
+                            1 * u.TeV,
+                            unique_azimuths[0],
+                            unique_zeniths[0],
+                            0,
+                            0 * u.deg,
+                        )  # Initialize grid point with azimuth and zenith
+                    else:
+                        _logger.warning(
+                            "Multiple unique values found for azimuth or zenith. "
+                            "The grid point will be set based on the first unique values."
+                        )
+                else:
+                    _logger.warning(
+                        f"Grid point already set to: {self.grid_point}. "
+                        "Overwriting with new values from file."
+                    )
+
+                    self.grid_point = (
+                        1 * u.TeV,
+                        unique_azimuths[0],
+                        unique_zeniths[0],
+                        0,
+                        0 * u.deg,
+                    )
+                    _logger.info(f"New grid point values: {self.grid_point}")
+
+        except FileNotFoundError as e:
+            error_message = f"Error loading file {self.file_path}: {e}"
+            _logger.error(error_message)
+            raise FileNotFoundError(error_message) from e
+        return data
+
+    def create_bin_edges(self):
+        """
+        Create unique energy bin edges.
+
+        Returns
+        -------
+        bin_edges : array
+            Array of unique energy bin edges.
+        """
+        bin_edges_low = self.data["bin_edges_low"]
+        bin_edges_high = self.data["bin_edges_high"]
+        bin_edges = np.concatenate([bin_edges_low, [bin_edges_high[-1]]])
+        return np.unique(bin_edges)
+
+    def compute_histogram(self, event_energies_reco, bin_edges):
+        """
+        Compute histogram for triggered events.
+
+        Parameters
+        ----------
+        event_energies_reco : array
+            Array of energies of the observed events.
```

Comment:
`Array of reconstructed energy per event`

Reply:
done